### ISSUE TITLE
Under certain circumstances, the parser changes the code if a negation is used

### ISSUE THREAD

**Issue by chris-l** [glayzzle/php-parser#75]
Hi! I was testing your PR by parsing several php files, and fixing differences.
But then I found something that is actually a bug of php-parser.

If you parse:

```php
if ($aaa && !$bbb) {
}
```

That code is an `if` that checks for `$aaa` and a negated `$bbb`.
Well, the generated AST is:

```json
{
  "kind": "program",
  "children": [
    {
      "kind": "if",
      "test": {
        "kind": "bin",
        "type": "&&",
        "left": {
          "kind": "unary",
          "type": "!",
          "what": {
            "kind": "variable",
            "name": "aaa",
            "byref": false,
            "curly": false
          }
        },
        "right": {
          "kind": "unary",
          "type": "!",
          "what": {
            "kind": "variable",
            "name": "aaa",
            "byref": false,
            "curly": false
          }
        }
      },
      "body": {
        "kind": "block",
        "children": []
      },
      "alternate": null,
      "shortForm": false
    }
  ],
  "errors": []
}
```

That is, is like if the code was being altered, changing the `$aaa` and the negated `$bbb`, to two negated `$aaa`, like this:

```php
if (!$aaa && !$aaa) {
}
```

**Comment by ichiriac** [comment]
Hi @chris-l,

Thanks for the report, I'll release a fix very soon.

**Comment by ichiriac** [comment]
fixed & released under 2.0.5

**Comment by chris-l** [comment]
Thanks a lot! With this I was able to finish the merge of your PR on php-unparser ;)
